### DISCUSSION TITLE
How to update/upgrade SFTPGO correctly

### DISCUSSION THREAD

**Discussion by rnenergy** [drakkan/sftpgo#904] [Q&A]
Sorry, there may already be an answer, but I couldn't find it. Tell me how to update the version of SFTPGO correctly, saving all the settings and files? The reinstallation of the deb package didn't work for me. (OS Debian, Ubuntu).

**Comment by drakkan** [comment]
Take a look [here](https://github.com/drakkan/sftpgo#upgrading). Installing the new deb should work if you are in a supported upgrade path, otherwise you need to install the intermediate versions first. What version are you trying to upgrade from? After the upgrade you should adapt your configuration file

**Reply by rnenergy** [reply]
It was 2.2.3 to 2.3.1

**Reply by drakkan** [reply]
> It was 2.2.3 to 2.3.1

it should work automatically. You only have to adapt the `sftpgo.json` file if you did some customizations in v2.2.3 but this will not prevent SFTPGo from starting. Please describe the exact steps you are doing and the SFTPGo output/logs

**Reply by rnenergy** [reply]
Ok. I'll try to do it in my test installation. Because the problematic installation has already been removed.

**Reply by rnenergy** [reply]
Thank you a lot.
